### ISSUE TITLE
[data grid] Actions column should not contain Menu/MenuItems if only 1 MenuItem is shown.

### ISSUE THREAD

**Issue by hungmanhle** [mui/mui-x#19902]
### Steps to reproduce

Steps:
1. Open this link to live example: https://codesandbox.io/embed/glxhkc?module=/src/Demo.tsx&fontsize=12 (Note this codesandbox is pulled from the latest MUI documentation: https://mui.com/x/react-data-grid/column-definition/)
2. Use screenreader (like VoiceOver)
3. Navigate to any one of the `...` buttons in the actions column.

### Current behavior

When there is a single button in the actions column, there is an unnecessary `Menu` and `MenuItem` components. This leads the screenreader to read the `...` button as a `MenuItem`.

This behaviour is what we want for multiple items in the actions column but is incorrect when there is only a single item.

<img width="932" height="506" alt="Image" src="https://github.com/user-attachments/assets/04b85912-03ca-4d20-b90d-69bebe78cae0" />

### Expected behavior

The `...` button should have the role of `button` instead of `menuitem`. There should not be a `role='menu'` wrapping around the button.

This applies to if there is only a single `action` in the `coldef`, but also should also apply if there are multiple `actions` in the `coldef` if all actions have `showInMenu` prop.

### Context

This poses a small accessibility issue, screenreaders will read this as a menu/menuitems when it's infact just a single button.

### Your environment

<details>
  <summary><code></code></summary>
System:
  OS: macOS 26.0
Browsers:
  Chrome: 141.0.7390.66
Using latest MUI package.

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**: datagrid, actions, column, menu, menuitem, a11y, accessibility

**Comment by MBilalShafi** [comment]
Other users also reported issues with the actions column API, and the team is working on an alternate API for the actions' column.

Could you check the in-progress PR and report if it could be a possible fix in your case? https://github.com/mui/mui-x/pull/15041

**Comment by hungmanhle** [comment]
@MBilalShafi checked the preview link and it does not resolve this issue. 

Each `GridActionsCell.Item` within the `GridActionsCell` is still a button with `role='menuitem'` which is correct when each there are multiple `GridActionsCell.Item`.

If there is a single `GridActionsCell.Item` It still renders a menu with a single menuitem.
If we remove `GridActionsCell` and use only a single `GridActionsCell.Item` it still renders the `GridActionsCell.Item` as a button with a `role='menuitem'` even though the `role='menu'` div provided by `GridActionsCell` is no longer in the markup.

So behaviour is the same as with previous API

**Comment by michelengelen** [comment]
@MBilalShafi gentle ping

**Comment by MBilalShafi** [comment]
> [@MBilalShafi](https://github.com/MBilalShafi) checked the preview link and it does not resolve this issue.
> 
> Each `GridActionsCell.Item` within the `GridActionsCell` is still a button with `role='menuitem'` which is correct when each there are multiple `GridActionsCell.Item`.
> 
> If there is a single `GridActionsCell.Item` It still renders a menu with a single menuitem. If we remove `GridActionsCell` and use only a single `GridActionsCell.Item` it still renders the `GridActionsCell.Item` as a button with a `role='menuitem'` even though the `role='menu'` div provided by `GridActionsCell` is no longer in the markup.
> 
> So behaviour is the same as with previous API

As per https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/:

> The element that opens the menu has role [button](https://w3c.github.io/aria/#button).

So, I agree that the current behavior is incorrect and could be fixed by a straightforward change:

```diff
diff --git a/packages/x-data-grid/src/components/cell/GridActionsCell.tsx b/packages/x-data-grid/src/components/cell/GridActionsCell.tsx
index 2120006d73..925e85af65 100644
--- a/packages/x-data-grid/src/components/cell/GridActionsCell.tsx
+++ b/packages/x-data-grid/src/components/cell/GridActionsCell.tsx
@@ -215,7 +215,7 @@ function GridActionsCell(props: GridActionsCellProps) {
           aria-haspopup="menu"
           aria-expanded={open}
           aria-controls={open ? menuId : undefined}
-          role="menuitem"
+          role="button"
           size="small"
           onClick={toggleMenu}
           touchRippleRef={handleTouchRippleRef(buttonId)}
```

Marked it as a good first issue. Feel free to open up a PR for the fix if you want.

**Comment by hungmanhle** [comment]
Opened https://github.com/mui/mui-x/pull/20219 @MBilalShafi

**Comment by cherniavskii** [comment]
Looks like expected behavior to me, see https://github.com/mui/mui-x/pull/20219#discussion_r2516030718 and https://github.com/mui/mui-x/pull/20219#issuecomment-3518984091

Feel free to reopen and elaborate if you disagree
